webextension/js/popup.js

```diff
@@ -141,6 +141,65 @@ const Logic = {
     return this._currentIdentity;
   },
 
+  cookieStoreId(userContextId) {
+    return `firefox-container-${userContextId}`;
+  },
+
+  _containerTabIterator(userContextId, cb) {
+    browser.tabs.query({
+      cookieStoreId: Logic.cookieStoreId(userContextId)
+    }).then((tabs) => {
+      tabs.forEach((tab) => {
+        cb(tab);
+      });
+    }).catch((e) => {throw e;});
+  },
+
+  _containers(userContextId) {
+    return browser.tabs.query({
+      cookieStoreId: Logic.cookieStoreId(userContextId)
+    });
+  },
+
+  sendTelemetryPayload(message = {}) {
+    if (!message.event) {
+      throw new Error("Missing event name for telemetry");
+    }
+    message.method = "sendTelemetryPayload";
+    browser.runtime.sendMessage(message);
+  },
+
+  removeIdentity(userContextId) {
+    const eventName = "delete-container";
+    if (!userContextId) {
+      return Promise.reject("removeIdentity must be called with userContextId argument.");
+    }
+
+    this.sendTelemetryPayload({
+      event: eventName,
+      userContextId
+    });
+
+    const removeTabsPromise = Logic._containers(userContextId).then((tabs) => {
+      const tabIds = tabs.map((tab) => tab.id);
+      return browser.tabs.remove(tabIds);
+    });
+
+    return removeTabsPromise.then(() => {
+      const removed = browser.contextualIdentities.remove(Logic.cookieStoreId(userContextId));
+      // Send delete event to webextension/background.js
```

Comment:
Nah just looked into this... I know this is an old PR however the eventName was reused but the key was different. In the latest PR. I have moved more code to use the "method" key which should make it more consistent rather than type. I also made the methodNames all camelcase rather than the telemetry which is dasherized.